src/ring_theory/mv_polynomial/basic.lean

```diff
@@ -117,3 +117,18 @@ rfl
 end degree
 
 end mv_polynomial
+
+/-- The monomials form a basis on `polynomial R`. -/
+noncomputable def polynomial.basis_monomials : basis ℕ R (polynomial R) :=
+((mv_polynomial.basis_monomials punit R).reindex
+  (finsupp.equiv_fun_on_fintype.trans (equiv.punit_arrow_equiv ℕ))).map
+  (mv_polynomial.punit_alg_equiv R).to_linear_equiv
+
+@[simp] lemma polynomial.coe_basis_monomials :
+  (polynomial.basis_monomials R : ℕ → polynomial R) = λ s, polynomial.monomial s 1 :=
+begin
+  apply funext,
+  intro n,
+  simpa [polynomial.basis_monomials, finsupp.equiv_fun_on_fintype] using
```

Comment:
This `simpa` is quite slow, I don't know how to make it faster.

Reply:
`@[simps]` does indeed produce `finsupp.equiv_fun_on_fintype_symm_apply`. Let me see if it helps.

Reply:
Does it produce it with the right statement?

Reply:
Yes, it is exactly the statement you wrote

Reply:
```
[simps] > adding projection finsupp.equiv_fun_on_fintype_symm_apply_to_fun:
        > ∀ {α : Type u_1} {M : Type u_5} [_inst_1 : has_zero M] [_inst_2 : fintype α] (f : α → M) (ᾰ : α),
  ⇑(⇑(equiv_fun_on_fintype.symm) f) ᾰ = f ᾰ
```